### PR TITLE
Swift 4 migration

### PR DESCRIPTION
**Pull request by artemnovichkov**
**Issue:** [Link](https://github.com/swift-xcode/xcodeproj/issues/82)

### Short description
Migrate the project to Swift 4

### Solution
Build the project with Xcode 9, fix all warnings and errors and update SPM manifest syntax.

### Implementation

- [x] Update `Package.swift` file. Please, check versions of dependencies. In my opinion, it's good practice to update dependencies to next major version.
- [x] Fix new syntax errors in `sorted(by:)` and `NSTextCheckingResult.range(at:)` functions.

### GIF
Smile

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,8 +51,13 @@ Add the dependency in your `Package.swift` file:
 let package = Package(
     name: "myproject",
     dependencies: [
-        .Package(url: "https://github.com/swift-xcode/xcodeproj.git", majorVersion: 0, minor: 0)
-    ]
+        .package(url: "https://github.com/swift-xcode/xcodeproj.git", .upToNextMajor(from: "0.0.0")),
```
Comment (pepibumur) [inline README.md]: We can update this to "0.2.0" *(the last version)*

**Comment by pepibumur** [comment]
@artemnovichkov do you have access to create branches on this repo?

**Comment by artemnovichkov** [comment]
One important remark: is it correct format of `Package.swift`? I saw [this manifest](https://github.com/Bouke/SRP/blob/master/Package.swift) and I'm not sure my version is correct. What do you think?

**Comment by pepibumur** [comment]
It is @artemnovichkov. Once travis passes we can merge it 😛

**Comment by yonaskolb** [comment]
You may to define `products` to be able to consume this framework. If you leave them out, I'm not sure if they are auto generated

**Comment by artemnovichkov** [comment]
@pepibumur I'm not sure, I forked the repo.

**Comment by yonaskolb** [comment]
`.swift-version` will need to be updated as well. Travis will probably fail without it

**Comment by pepibumur** [comment]
Yep, the bash script used to install swift uses `.swift-version` to determine which version needs to be installed.

**Comment by artemnovichkov** [comment]
@pepibumur have you checked CI errors?
`We don't have build instructions for 4.0.`
I guess swiftenv doesn't support Swift 4.

**Comment by pepibumur** [comment]
It looks like it only has `4.0-dev` [link](https://github.com/kylef/swiftenv/tree/master/share/swiftenv-build). Can you try that one?

**Comment by artemnovichkov** [comment]
I should update `.swift-version` file, right?

**Comment by pepibumur** [comment]
Yes @artemnovichkov, and just use `4.0-dev`.

**Comment by pepibumur** [comment]
Another thing that we can do is remove the swiftenv step and use the version that comes with Xcode 9. That should also work. By doing that, the execution should be faster on CI.

**Comment by artemnovichkov** [comment]
I have integrated buddybuild to my open-source project recently, and everything works w/o swiftenv

**Comment by pepibumur** [comment]
Then let's try that. Just remove the line from `.travis.yml` that installs Swift. That should work.

**Comment by pepibumur** [comment]
I think we need to update also the `Tests` folder structure @artemnovichkov

**Comment by artemnovichkov** [comment]
@pepibumur I did it!

**Comment by pepibumur** [comment]
Bum! Ready to merge. Nice job @artemnovichkov

**Pull request merged**